### PR TITLE
Add startup logging

### PR DESCRIPTION
**Pull request by SammyK**
### Description

This PR adds startup logs that contain a JSON-encoded string of configuration and diagnostic information. The JSON can be provided to the support team to help identify configuration issues. The log message is only generated on process startup and can be disabled by setting `DD_TRACE_STARTUP_LOGS=0`.

#### Example startup message

```
[2020-07-01T17:42:50Z] DATADOG TRACER CONFIGURATION - {"agent_error":"Couldn't connect to server","ddtrace.request_init_hook_reachable":false,"date":"2020-07-01T17:42:50Z","os_name":"Linux 49b1cb4bdd12 4.19.76-linuxkit #1 SMP Tue May 26 11:42:35 UTC 2020 x86_64","os_version":"4.19.76-linuxkit","version":"1.0.0-nightly","lang":"php","lang_version":"7.4.5","env":null,"enabled":true,"service":null,"enabled_cli":false,"agent_url":"https://localhost:8126","debug":false,"analytics_enabled":false,"sample_rate":1.000000,"sampling_rules":null,"tags":null,"service_mapping":null,"distributed_tracing_enabled":true,"priority_sampling_enabled":true,"dd_version":null,"architecture":"x86_64","sapi":"cgi-fcgi","ddtrace.request_init_hook":null,"open_basedir_configured":false,"uri_fragment_regex":null,"uri_mapping_incoming":null,"uri_mapping_outgoing":null,"auto_flush_enabled":false,"generate_root_span":true,"http_client_split_by_domain":false,"measure_compile_time":true,"report_hostname_on_root_span":false,"traced_internal_functions":null,"auto_prepend_file_configured":false,"integrations_disabled":null,"enabled_from_env":true,"opcache.file_cache":null}
```

When debug mode is enabled (`DD_TRACE_DEBUG=1`), failed diagnostic tests will also generate log messages:

```
[2020-07-01T17:35:25Z] DATADOG TRACER DIAGNOSTICS - agent_error: Couldn't connect to server
[2020-07-01T17:35:25Z] DATADOG TRACER DIAGNOSTICS - ddtrace.request_init_hook_reachable: false
```

#### PHP Info

The JSON string can also be obtained from the PHP info page and the diagnostic information will be displayed in a separate table to help diagnose common issues.

ddtrace PHP info page

From the CLI SAPI, the info can be obtained from `php --ri=ddtrace`.

```
ddtrace


Datadog PHP tracer extension
For help, check out the documentation at https://docs.datadoghq.com/tracing/languages/php/
(c) Datadog 2020

Datadog tracing support => enabled
Version => 1.0.0-nightly
DATADOG TRACER CONFIGURATION => {"date":"2020-07-01T17:43:54Z","os_name":"Linux 49b1cb4bdd12 4.19.76-linuxkit #1 SMP Tue May 26 11:42:35 UTC 2020 x86_64","os_version":"4.19.76-linuxkit","version":"1.0.0-nightly","lang":"php","lang_version":"7.4.5","env":null,"enabled":true,"service":null,"enabled_cli":false,"agent_url":"https://localhost:8126","debug":false,"analytics_enabled":false,"sample_rate":1.000000,"sampling_rules":null,"tags":null,"service_mapping":null,"distributed_tracing_enabled":true,"priority_sampling_enabled":true,"dd_version":null,"architecture":"x86_64","sapi":"cli","ddtrace.request_init_hook":null,"open_basedir_configured":false,"uri_fragment_regex":null,"uri_mapping_incoming":null,"uri_mapping_outgoing":null,"auto_flush_enabled":false,"generate_root_span":true,"http_client_split_by_domain":false,"measure_compile_time":true,"report_hostname_on_root_span":false,"traced_internal_functions":null,"auto_prepend_file_configured":false,"integrations_disabled":null,"enabled_from_env":true,"opcache.file_cache":null,"agent_error":"Couldn't connect to server","ddtrace.request_init_hook_reachable":false}

                               Diagnostics
agent_error => Couldn't connect to server
ddtrace.request_init_hook_reachable => false

Directive => Local Value => Master Value
ddtrace.disable => Off => Off
...
```

The JSON string can also be accessed at runtime from `\DDTrace\startup_logs()`.

```php
echo \DDTrace\startup_logs() . PHP_EOL;
```

### Readiness checklist
- [x] (only for Members) Changelog has been added to the appropriate release draft. Create one if necessary.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the appropriate release draft. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: src/ext/ddtrace.c [hunk 1]
```diff
@@ -405,6 +416,100 @@ static PHP_RSHUTDOWN_FUNCTION(ddtrace) {
...
+                    break;
+            }
+        }
+        zend_hash_move_forward_ex(ht, &pos);
+    }
+#endif
+
+    if (zend_hash_num_elements(ht) == 0) {
+        php_info_print_table_row(2, "All diagnostic checks", "passed");
+    }
```
Comment (morrisonlevi) [inline src/ext/ddtrace.c:505]: Would it be better to always have this row and have passed/did not pass?

File: src/ext/php5/startup_logging.c [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        (ini_len == 2 && strcasecmp(ini, "on") == 0)) {
+        return 1;
+    } else {
+        return atoi(ini) != 0;
+    }
+}
+
+static void _dd_get_agent_url(char *buf) {
+    char *host = get_dd_agent_host();
+    sprintf(buf, "https://%s:%d", host, (int)get_dd_trace_agent_port());
```
Comment (morrisonlevi) [inline src/ext/php5/startup_logging.c:86]: Are you about http vs https here?

File: src/ext/ddtrace.c (hunk 1 above)
Comment (SammyK) [reply src/ext/ddtrace.c]: Good point. I updated this. 👍

File: src/ext/php5/startup_logging.c (hunk 2 above)
Comment (SammyK) [reply src/ext/php5/startup_logging.c]: Nice catch. This is actually supposed to be `http` and I updated the PR. Once #926 is merged, we could expose a public function to generate the URL for consistency.

**Pull request merged**